### ISSUE TITLE
Hot Reload remembers the last reload if not saved,and will keep printing the task that is not given to it

### ISSUE THREAD

**Issue by davord45** [fian46/addons-btree#29]
Hi fian46,i found a bug in hot reload,i tried to print "first" than change it to "second",that worked like in the video,but than i tested i you dont press save it will keep printing "second" with "first" even though first is its parameter.I tried fixing it by saving and it keeps printing the same thing.
![image](https://user-images.githubusercontent.com/70846358/94114350-86269f80-fe48-11ea-86a7-294345c89683.png)

**Comment by fian46** [comment]
did you connect to debugger first ?

**Comment by fian46** [comment]
it will not working if you nit connect to debugger. if so it will hint you with message BT Hot reload in hint bar

**Comment by fian46** [comment]
the hint message only say saving data so no hot reload performed

**Comment by davord45** [comment]
Yeah i did that,but it keeps remembering the last action and prints that until i hot reload again,like it almost runs paralel to the startup parameter,yeah im checking when i increase the wait it prints it twice,and when i hot reload it prints twice hot reload message

**Comment by davord45** [comment]
Yeah it keeps the first parameter on the first line and the hot reloaded one on the second

**Comment by fian46** [comment]
Can you upload small reproduce project about this issue ? Also with the step too ?

**Comment by fian46** [comment]
Also it will reload until you press ctr + s or press save in editor. My bad not tell you that.

**Comment by davord45** [comment]
I will test it when i get back from dinner,but why does it remember the reload after the runtime is over?

**Comment by davord45** [comment]
Shouldnt it be going back to what is in tree editor

**Comment by fian46** [comment]
I am not really sure, i tried to recreate the node in tour screenshot but it not behaving like your case

**Comment by davord45** [comment]
I recorded the bug in video so that you can see everything
https://youtu.be/a0B-qNJP4kU

**Comment by davord45** [comment]
Now i get it printing first even when i restart the editor

**Comment by fian46** [comment]
Wow that looks bad. Thanks for the report. I will looking for it. But can you send me just the scene and the script ? I need to looking more details.

**Comment by davord45** [comment]
Here you go,hope you can reproduce it.
http://www.mediafire.com/file/pjd41ttpotdeug9/HotReloadBug.zip/file

**Comment by fian46** [comment]
yeah this is really funny. it looks like your parameter in print is double "first enter first" that is why it print 2 "first" every tick. when you edit "first" to "second" you only change it to "second enter first". i am still consider this as usability problem really, because user still can't really see every parameter value in full. this is not connected to any hot reload problem. but i don't know how to fix it right now. maybe make the text area larger for parameter input ?

**Comment by davord45** [comment]
How do you mean larger?How would that fix the problem?

**Comment by davord45** [comment]
Tried resizing the node if thats what you mean,but it did not fix it

**Comment by davord45** [comment]
What is a double parameters,how could that be when i only send one parameter?You only have parameters for Strings and Numbers,is this like an array?

**Comment by fian46** [comment]
No in your scene file the parameter is 0 is "first enter first". Please scroll down your mouse when edit the paramater. The enter is real enter character btw not enter like text. There is 2 line in your parameter 0

**Comment by fian46** [comment]
First line is "first" and second line is "first"
When you edit it you assuming only one line but there is two. And the second line is unedited because you cant see it.

**Comment by davord45** [comment]
Ah must be because i pressed enter

**Comment by fian46** [comment]
Yes it fixed in my side. Maybe i can upload some video ?

**Comment by davord45** [comment]
Yeah that was the case,didnt see you have a mini scroll bar on the node

**Comment by davord45** [comment]
But it still hot reloads twice the first time

**Comment by fian46** [comment]
That is fine it is design flaw. Maybe for large text input it need some popup to enter the text for better experience.

**Comment by davord45** [comment]
Actually when i press Save in Editor it does only once,must be a keyboard thing

**Comment by fian46** [comment]
> But it still hot reloads twice the first time

What do you mean ?

**Comment by fian46** [comment]
Oh that double print hot reload thing. Yes it is some extra unnecessary reload because the editor intercept the same event as the plugins do. It is unfortunate but that will not changed the operation.

**Comment by davord45** [comment]
It hot reloads on ctrl+s press and release

**Comment by davord45** [comment]
Ok i found another issue,when you make two exact same trees only name change,and hot reload a parameter from one it will change it in both.

**Comment by fian46** [comment]
Yeah that's another issue in id_generator i just realized that can happen. I will look into it tomorrow 😁

**Comment by davord45** [comment]
Get some rest dude,you deserved it.

**Comment by davord45** [comment]
You fixed the issue,thanks.
